src/Ontology/Address.cpp

```diff
@@ -0,0 +1,87 @@
+// Copyright © 2017-2019 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "Hash.h"
+#include "Address.h"
+#include "HexCoding.h"
+
+#include <stdexcept>
+
+#include <TrezorCrypto/sha2.h>
+#include <TrezorCrypto/base58.h>
+#include <TrezorCrypto/ripemd160.h>
+
+using namespace TW::Ontology;
+
+Address::Address(const PublicKey &publicKey) {
+    std::vector<uint8_t> builder(publicKey.bytes);
+    builder.insert(builder.begin(), 0x21);
+    builder.push_back(0xAC);
+    auto data = toScriptHash(builder);
+    std::copy(data.begin(), data.end(), zero.begin());
+}
+
+
+Address::Address(const std::string &b58Address) {
+    size_t dataSz = 25;
```

Comment:
```suggestion
    size_t dataSize = 25;
```

Reply:
I had changed in newest code, maybe you need to resolve conflicts first. then, my newest code you can see.